### PR TITLE
bug: fix sender ioc error

### PR DESCRIPTION
**Pull request by lasrinnil**
https://portal.azure.com/#view/AppInsightsExtension/DetailsV2Blade/ComponentId~/%7B%22SubscriptionId%22%3A%223c13ff3d-aad8-4ee8-94c6-8a938ace9699%22%2C%22ResourceGroup%22%3A%22rg-DataHub-SharedResouces-U-001%22%2C%22Name%22%3A%22appi-shared-sharedres-u-001%22%2C%22LinkedApplicationType%22%3A0%2C%22ResourceId%22%3A%22%252Fsubscriptions%252F3c13ff3d-aad8-4ee8-94c6-8a938ace9699%252FresourceGroups%252Frg-DataHub-SharedResouces-U-001%252Fproviders%252FMicrosoft.Insights%252Fcomponents%252Fappi-shared-sharedres-u-001%22%2C%22ResourceType%22%3A%22microsoft.insights%252Fcomponents%22%2C%22IsAzureFirst%22%3Afalse%7D/DataModel~/%7B%22eventId%22%3A%2229051c52-4dec-11ed-9464-000d3aac8edc%22%2C%22timestamp%22%3A%222022-10-17T07%3A19%3A54.050Z%22%2C%22cacheId%22%3A%2206061ce8-3f98-48b5-812f-4dfe06d5b0fa%22%2C%22eventTable%22%3A%22exceptions%22%2C%22timeContext%22%3A%7B%22durationMs%22%3A86400000%2C%22endTime%22%3A%222022-10-17T07%3A26%3A42.168Z%22%7D%7D

### CODE REVIEW

**Review by johevemi — APPROVED** [review]

File: source/dotnet/Services/Sender/Program.cs [hunk 1]
```diff
@@ -81,6 +87,7 @@ private static void MiddlewareServices(IServiceCollection services)
         services.AddScoped<CorrelationIdMiddleware>();
         services.AddScoped<IIntegrationEventContext, IntegrationEventContext>();
         services.AddScoped<IntegrationEventMetadataMiddleware>();
+        services.AddScoped<IntegrationEventMetadataMiddleware>();
```
Comment (JesperGomes) [inline source/dotnet/Services/Sender/Program.cs:90]: Why is this added twice?

Comment (lasrinnil) [reply source/dotnet/Services/Sender/Program.cs:90]: nice catch!

**Review by JesperGomes — APPROVED** [review]

File: source/dotnet/Services/IntegrationTests/Hosts/SenderIntegrationTestHost.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    private static void ConfigureServices(IServiceCollection serviceCollection)
+    {
+        serviceCollection.Replace(ServiceDescriptor.Singleton<ServiceBusClient, MockedServiceBusClient>());
+        serviceCollection.Replace(ServiceDescriptor.Singleton(provider =>
+        {
+            var mock = new Moq.Mock<IServiceBusClientFactory>();
+            mock.Setup(x => x.Create()).Returns(provider.GetRequiredService<ServiceBusClient>());
+            return mock.Object;
+        }));
```
Comment (BjarkeMeier) [inline source/dotnet/Services/IntegrationTests/Hosts/SenderIntegrationTestHost.cs:95]: Don't these potentially prevent the detection of missing registrations? Example: What if the `ServiceBusClientFactory` have a dependency that's not added to the container?

File: source/dotnet/Services/IntegrationTests/Hosts/SenderIntegrationTestHost.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        serviceCollection.AddScoped<PeekEndpoint>();
```
Comment (BjarkeMeier) [inline source/dotnet/Services/IntegrationTests/Hosts/SenderIntegrationTestHost.cs:96]: This is specific only to your one single test. I would add it in the test.
